### PR TITLE
DEPR: inplace argument in set_axis

### PR DESCRIPTION
**Pull request by lukemanley**
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/v2.0.0.rst` file if fixing a bug or adding a new feature.

introduced in #48130

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -754,45 +720,24 @@ def set_axis(
...
-            )
-        else:
-            inplace = False
-
-        if inplace:
-            if copy is True:
-                raise ValueError("Cannot specify both inplace=True and copy=True")
-            copy = False
-        elif copy is lib.no_default:
+        if copy is lib.no_default:
```
Comment (jbrockmendel) [inline pandas/core/generic.py:737]: i think copy can just default to True now

Comment (lukemanley) [reply pandas/core/generic.py]: fixed, thx

**Review by mroeschke — APPROVED** [review]
LGTM. Merge when ready @jbrockmendel

**Review by jbrockmendel — APPROVED** [review]

**Comment by jbrockmendel** [comment]
thanks @lukemanley

**Pull request merged**